packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts

```diff
@@ -503,6 +511,25 @@ export class StripeHandler {
         metricsContext,
       } = request.payload as Record<string, string>;
 
+      // Make sure to clean up any subscriptions that may be hanging with no payment
+      const existingSubscription =
+        this.stripeHelper.findCustomerSubscriptionByPlanId(customer, priceId);
+      if (
+        existingSubscription &&
+        existingSubscription.status === 'incomplete'
+      ) {
+        this.stripeHelper.cancelSubscription(existingSubscription.id);
```

Comment:
Do we want to wait on this to make sure its actually removed before proceeding?

Reply:
Was following a similar pattern to https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/lib/payments/stripe.ts#L582.

Regardless, have updated this to wait for the promise to resolve.

Reply:
Ah, that other one should prolly be awaited as well.